### ISSUE TITLE
[receiver/hostmetrics] Remove shared from system.memory.state values

### ISSUE THREAD

**Issue by ChrsMark** [open-telemetry/opentelemetry-collector-contrib#32712]
### Component(s)

receiver/hostmetrics

### Describe the issue you're reporting

If https://github.com/open-telemetry/semantic-conventions/pull/933 is merged we will need to change the receiver implementation accordingly.

ref: https://github.com/open-telemetry/semantic-conventions/issues/522

**Comment by crobert-1** [comment]
Hopefully not taking this out of context, but a [code owner's response](https://github.com/open-telemetry/semantic-conventions/issues/522#issuecomment-2047653901) on another issue seems to agree with this decision as well. 

_Removing `needs triage`._

**Comment by rogercoll** [comment]
Still important for v1. Once [slab memory is available in the semantic conventions](https://github.com/open-telemetry/semantic-conventions/pull/1078), we could implement the `shared` + `slab` removal under the same feature gate.

**Comment by osullivandonal** [comment]
I was looking at this with the idea of picking it up, but after reviewing https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/d833a7e12e5a3262616fab2200c457e196e9123b/receiver/hostmetricsreceiver/internal/scraper/memoryscraper/metadata.yaml?plain=1#L17 `shared` is not part of the metadata file at all, I think we can close this? @crobert-1 @ChrsMark @rogercoll

**Comment by ChrsMark** [comment]
Sounds good to me. @braydonk please confirm that we don't miss anything here.

**Comment by rogercoll** [comment]
Good point, maybe we can rename the issue to add the new [system.memory.shared](https://github.com/ChrsMark/semantic-conventions/blob/main/docs/system/system-metrics.md#metric-systemmemoryshared) metric instead.

**Comment by osullivandonal** [comment]
I'd like to work on this! Could we get the issue name updated to reflect adding the new `system.memory.shared` metric? Cheers!

cc. @ChrsMark
